game-app/game-core/src/main/java/games/strategy/engine/framework/AbstractGame.java

```diff
@@ -48,6 +51,9 @@ public abstract class AbstractGame implements IGame {
   @Nullable private IDisplay display;
   @Nullable private ISound sound;
 
+  // TODO ensure ResourceLoader is never null
```

Comment:
Good question I was thinking about some sort of placeholder resource loader until a game-specific resource loader is put in place?

Reply:
IIRC there is almost always a resource loader, if no game-specific resource loader is in place then there is a resource loader to the assets folder which pulls the graphics needed before a game.

I suspect the nullability comes from issues with data initialization and perhaps circular dependencies. It may also come as a poor way to do garbage collection with object references being nulled out so that they are collected properly